components/relay/src/lib.rs

```diff
@@ -0,0 +1,350 @@
+mod error;
+
+pub use error::{Error, Result};
+
+use serde::{Deserialize, Serialize};
+use url::Url;
+use viaduct::{header_names, Method, Request};
+
+pub struct RelayClient {
+    server_url: String,
+    auth_token: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RelayAddress {
+    pub mask_type: String,
+    pub enabled: bool,
+    pub description: String,
+    pub generated_for: String,
+    pub block_list_emails: bool,
+    pub used_on: Option<String>,
+    pub id: i64,
+    pub address: String,
+    pub domain: i64,
+    pub full_address: String,
+    pub created_at: String, // Use String for timestamps for now (or chrono types later)
```

Comment:
this seems unfortunate, but I guess doesn't need to block here. Also it might make sense to add docstrings here as uniffi etc is capable of generating user-facing docs with those strings (but again, later is fine)